app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,68 @@
+class TweetsController < ApplicationController
+ get '/tweets' do
+   if logged_in?
+     @tweets = Tweet.all
+     erb :'/tweets/tweets'
+   else
+     redirect to "/login"
+   end
+ end
+
+ get '/tweets/new' do
+   if logged_in?
+     @user = User.find_by_id(session[:user_id])
+     erb :'/tweets/create_tweet'
+   else
+     redirect to "/login"
+   end
+ end
+
+ get '/tweets/:id' do
+   if logged_in?
+     @tweet = Tweet.find_by_id(params[:id])
+     erb :'/tweets/show_tweet'
+   else
+     redirect to "/login"
+   end
+ end
+
+ post '/tweets' do
+   if params["content"].empty?
+     redirect to "/tweets/new"
+   else
+     @tweet = Tweet.create(:content => params["content"], :user_id => session[:user_id])
+     redirect to "/tweets"
+   end
+ end
+
+ get '/tweets/:id/edit' do
+   if logged_in?
+     @tweet = Tweet.find_by_id(params[:id])
+     erb :'/tweets/edit_tweet'
+   else
+     redirect to "/login"
+   end
+ end
+
+ patch '/tweets/:id' do
+    @tweet = Tweet.find_by_id(params[:id])
+   if params["content"].empty?
+     redirect to "/tweets/#{@tweet.id}/edit"
+   else
+     @tweet.update(content: params[:content])
+     @tweet.save
+     redirect to "/tweets/#{@tweet.id}"
+   end
+end
+
+ delete '/tweets/:id/delete' do
+    @tweet = Tweet.find_by_id(params[:id])
+    #binding.pry
+    if session[:user_id] == @tweet.user_id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.